uefi/visitor.go

```diff
@@ -0,0 +1,28 @@
+package uefi
+
+// Visitor represents an operation which can be applied to the Firmware. To
+// implement a visitor, implement a method for each concrete Firmware type you
+// want to visit. The remaining nodes can be recursed over automatically with:
+//
+//     func (v *Find) VisitFV(fv *uefi.FirmwareVolume) error {
+//             return fv.ApplyChildren(v)
+//     }
+//
+// Or pruned with:
```

Comment:
I don't see the logic for pruning? 

In any case, I think pruning should rather happen with something like `uefi.ErrRemove`. People might return `nil` for "do nothing" or "I successfully collected some metadata from this item".

Reply:
Returning nil will not modify the tree at all. Returning `nil` does nothing which includes not recusing over children.

Maybe pruning is the wrong word. I got it from the CS idea of "pruning a search space".